server/config/index.js

```diff
@@ -23,8 +28,17 @@ var pipeline = [
 
     function mergeDefaultAndFileSettings(context) {
         context.result = _.merge(context.defaults, context.file);
-    }
+    },
 
+    function getCommandLineArgs(context) {
+        if (program.port) {
+            console.log("BEFORE");
+            console.log(context);
+            context.result.port = program.port;
```

Comment:
@hhaidar Yo should I do something like above with the merging? I noticed something was weird with the output because context.defaults.port was being changed when you changed context.result.port.